### ISSUE TITLE
Create TestFlight versions of retroarch nightly arm64 without flycast or play

### ISSUE THREAD

**Issue by Iamthebestintheworld222** [libretro/RetroArch#17038]
I'm not using AltStore or sidestore because I am using only emulators from the App Store and TestFlight. I can't run Saturn, tg16, TGCD, supergrafx, DSI, PCFX, FBNeo and mame arcade games, 32x, sega cd, sg1000, msx, pc98, dos, x68k, intellivision, colecovision, odyssey2, channel f, amiga, c64, Vic-20 and more without RetroArch. It would be nice if RetroArch nightly came to TestFlight and removed flycast and play since those use jit.

**Comment by hizzlekizzle** [comment]
There is already a TestFlight build, but it is identical to the regular App Store, just released nightly: https://testflight.apple.com/join/A1LMA3Mx

**Comment by Iamthebestintheworld222** [comment]
There are a lot of cores missing.

**Comment by Iamthebestintheworld222** [comment]
The other cores should only be in the TestFlight versions before being added to the App Store version

**Comment by warmenhoven** [comment]
The TestFlight build is exactly the same as the App Store build.

**Comment by warmenhoven** [comment]
Here are the list of cores in the TestFlight and App Store builds:

https://github.com/libretro/RetroArch/blob/master/pkg/apple/update-cores.sh#L119

**Comment by Iamthebestintheworld222** [comment]
Here are all the core files
https://buildbot.libretro.com/nightly/apple/ios-arm64/latest/

**Comment by Iamthebestintheworld222** [comment]
No mame or yabause

**Comment by Iamthebestintheworld222** [comment]
No beetle supergrafx either

**Comment by warmenhoven** [comment]
That's right, the App Store and TestFlight versions do not include all possible cores, only the ones we maintain internally or have upstream approval to put on stores. But the list of cores for the App Store and TestFlight are identical.

**Comment by Iamthebestintheworld222** [comment]
No beetle PCFX either

**Comment by warmenhoven** [comment]
mednafen_supergrafx is in the TestFlight.

**Comment by Iamthebestintheworld222** [comment]
But since the Dylib cores are available, why not create a download core option like the pc versions?

**Comment by warmenhoven** [comment]
Apple rules. Executables have to be bundled in the IPA and can't be downloaded. The in-app purchases for other apps are all data files or unlocks; the executable code is pre-bundled in the original IPA.

**Comment by warmenhoven** [comment]
Many of our cores have explicit non-commercial licenses; charging would violate the licenses. Also it still wouldn't fix that we wouldn't get approval for distributing the cores through a storefront.

**Comment by Iamthebestintheworld222** [comment]
Is there a folder where i can put the dylib cores? It's probably hidden somewhere

**Comment by Iamthebestintheworld222** [comment]
Or a core dylib install option

**Comment by warmenhoven** [comment]
> Is there a folder where i can put the dylib cores? ... Or a core dylib install option

Not exactly. Binaries have to be inside the app bundle and Apple does an integrity check of the bundle before it will allow the app to execute. As a result, you can't add dylib files to the app after the app is installed.

If you side load, then you can add (or remove) cores from the side load before you install the IPA. An IPA is just a ZIP file, and you can edit the files that are in there before you install it. The side load version that's available on the buildbot has all of the cores embedded in it, so it really would only make sense to delete cores from it, and most people don't choose to do that.

**Comment by Iamthebestintheworld222** [comment]
Can't you only add beetle PC-FX and mame for compatibility

**Comment by warmenhoven** [comment]
Mame isn't going to happen, but FBNeo is there. I've put up a [PR](https://github.com/libretro/RetroArch/pull/17044) to add Beetle PC-FX.
